### ISSUE TITLE
Browser ignoring .border

### ISSUE THREAD

**Issue by lukeyouell** [tailwindcss/tailwindcss#435]
I've come across an issue where the browser is ignoring the `.border` class:

<img width="344" alt="image" src="https://user-images.githubusercontent.com/24975058/37904566-1f4914f6-30f4-11e8-9903-7160a23791ad.png">

This happens in both Chrome & Firefox.

**Comment by adamwathan** [comment]
What do you mean it's ignoring it? Can you share a project that demonstrates the issue along with both the expected result and the actual result? Not sure what I'm really looking at in that screenshot.

**Comment by adamwathan** [comment]
If you aren't using our `@tailwind preflight` reset, `.border` won't do anything on it's own as it only sets border-width and the browser's default border-style is `none`. So if you are layering just utilities on top of some existing CSS, you'll want to add `.border-solid` as well whenever working with borders.

**Comment by lukeyouell** [comment]
The border isn't being applied, the screenshot I provided is of the element when inspected.

Unfortunately I can't share the project, but I've attached a few more screenshots below:

This should have a one pixel border, code below screenshot:
<img width="161" alt="image" src="https://user-images.githubusercontent.com/24975058/37905306-5323b342-30f6-11e8-93a3-2d3e3b65cb0c.png">
```html
<a class="text-white no-underline border border-blue-light hover:border-white py-3 px-4 rounded" href="#">Contact</a>
```

Inspect element results:
<img width="348" alt="image" src="https://user-images.githubusercontent.com/24975058/37905375-91cdf526-30f6-11e8-85fa-f47a448e4d63.png">

**Comment by lukeyouell** [comment]
> If you aren't using our @tailwind preflight reset, .border won't do anything on it's own as it only sets border-width and the browser's default border-style is none. So if you are layering just utilities on top of some existing CSS, you'll want to add .border-solid as well whenever working with borders.

Could be a side-effect of using Purge CSS?

**Comment by lukeyouell** [comment]
Using `.border-solid` solves it

**Comment by adamwathan** [comment]
> Unfortunately I can't share the project, but I've attached a few more screenshots below:

You can make a new project that reproduces the issue though yeah? It's hard to troubleshoot a screenshot of dev tools.

Are you using Preflight or not?

**Comment by fedeTibaldo** [comment]
As far as I know is more of a design decision than a side effect of using
PurgeCSS.

Border style gets applied together with border color only once on
preflight. You can actually see these two attributes in the `tailwind.js`
configuration file. In this way, whenever you need a border, you tipically
only have to set the width.

**Comment by adamwathan** [comment]
Closing as not a bug or issue with Tailwind itself but happy to help you figure out where that style is coming from in your project if you can share a reproduction.

**Comment by HapLifeMan** [comment]
I am using your [laravel-preset](https://github.com/adamwathan/laravel-preset) and I I have the same issue in production mode on my server.

After a `npm run prod` on my server, I can see this in `app.css`:
```css
*,:after,:before{border:0}
```
Trying this on my Mac with `npm run prod` or `npm run watch` is fine:
```css
*,:after,:before{border:0 solid #dae1e7}
```

I referenced my own `tailwind.js` in the `postCss()` options on build:
```js
require('tailwindcss')('./tailwind.js'),
```

Do you still want something to reproduce or do you have enough informations?

**Comment by adamwathan** [comment]
Not sure why there would be a difference between local and production; best suggestion is to make sure you are using the same version of node and npm in both environments and see if the issue still persists.

I build all my Tailwind stuff on the server too with no issues, using node 8.11.1 and yarn 1.6.0 on the server I just checked.

**Comment by pusle** [comment]
> Using `.border-solid` solves it

Thank you for posting that. I have the same problem, but no purgeCSS.

**Comment by luisrudge** [comment]
I'm also seeing this issue in tailwindcss@next 🤔 I'm not using purgeCSS.

**Comment by theomjones** [comment]
I'm also seeing this issue on Chrome & Firefox, in a Vue app using 1.0.1.
No purgecss

**Comment by theomjones** [comment]
For me this was because I was using `@preflight` directive instead of `@base`. There's still a lot of unofficial documentation out there using that, so watch out!

**Comment by drinkmorewaters** [comment]
Got this issue currently using `@base`

**Comment by jasonday** [comment]
Note: `.border-solid` ignores specific borders such as `.border-b`, and applies a border to all sides of the object.

**Comment by mike-seekwell** [comment]
should `.border-solid` be in tailwindui components?

**Comment by muhsalaa** [comment]
why when using `.border-solid` all sides have borders even if I use `border-r-2`?  I do preflight reset.

**Comment by jasonday** [comment]
Why was this issue closed?

**Comment by adamwathan** [comment]
It's not an issue, everything works as expected. If you have a bug to report, please open a new issue and include a GitHub repository that reproduces the problem.

**Comment by ahmedhosny** [comment]
Adding preflight solved the exact same issue for me.

**Comment by nemeton-x** [comment]
Hello @jasonday , I had the similar issue `.border-b`, `divide-y` were the classes not working for me.
As you can see below.
![image](https://user-images.githubusercontent.com/14342057/105357565-04d6f200-5c1b-11eb-8d05-7f90c142faa5.png)

This was my **tailwind.config.js** looked like

![image](https://user-images.githubusercontent.com/14342057/105357678-29cb6500-5c1b-11eb-9325-26d6f83d13fe.png)

After making `default` => `DEFAULT`, it worked for me, Give it a try, it might work for you as well. Also keeping `DEFAULT` is what tailwind documentation recommends as well.

![image](https://user-images.githubusercontent.com/14342057/105358206-e02f4a00-5c1b-11eb-8eae-8fdc98a44377.png)


![image](https://user-images.githubusercontent.com/14342057/105358155-cdb51080-5c1b-11eb-8997-2a52be76c72e.png)

**Comment by RobRoseKnows** [comment]
This is currently an issue with the [Next.js example](https://github.com/vercel/next.js/blob/canary/examples/with-tailwindcss/pages/index.js). `border` is defined in the classNames for the anchor elements, but copying the code won't actually make the borders appear.

**Comment by ababakanian** [comment]
I have also experienced this, here is the code:
https://codesandbox.io/embed/bold-stonebraker-dw0ywo?fontsize=14&hidenavigation=1&theme=dark

hopefully I made a mistake somewhere

**Comment by Illia-S-dev** [comment]
@adamwathan Good afternoon. Could you help me, please? I have strange behaviour with border.
I use next and tailwind.

As you can see on the screen I add some styles (for border and padding-left).
But In the dev-tools there is only style for padding, not for border.

I have an additional project (just react and tailwind). Everything works well in this project.

I can't understand what I do wrong.

Any help would be appreciated.

![Screenshot 2023-10-17 at 10 55 57](https://github.com/tailwindlabs/tailwindcss/assets/66375193/ca341cfa-0f79-4e54-ab0d-1dbebc6d293c).

**Comment by muratdogan17** [comment]
@Illia-S-dev did you solve the issue you posted? I have the same one
